code/parachain/frame/fnft/src/lib.rs

```diff
@@ -471,14 +475,14 @@ pub mod pallet {
 	}
 
 	impl<T: Config> FinancialNft<AccountIdOf<T>> for Pallet<T> {
-		/// TODO (vim): Assess the probability of collision in generating accounts
-		///   with collection id type u128 and instance id type u128 this definitely collides
-		///   because the seed is longer than the account ID
 		fn asset_account(
 			collection: &Self::CollectionId,
 			instance: &Self::ItemId,
 		) -> AccountIdOf<T> {
-			T::PalletId::get().into_sub_account_truncating((collection, instance))
+			let protocol_collection_id =
+				T::CurrencyFactory::unique_asset_id_to_protocol_asset_id(*collection);
```

Comment:
What makes this collision free?

Reply:
The size of `protocol_collection_id` is `u32` (4 bytes), and the size of `instance` is `u64` (8 bytes). 
We have 20 bytes of space to create a unique seed, so the combination of both of these values puts us comfortably under that limit.

Reply:
I see, as explained offline, if we can find a way to enforce these sizes remain within bounds at compile time it would be great.

Reply:
After a conversation with Antonio, I've decided to go back to hashing
91811fae61299f12d3d06253da223a4762fed424